### PR TITLE
[DJFS-40] Querying the Transportation Database  

### PR DESCRIPTION
**Pull request by opeyemipeter1759**
## Task: `Name of given task`
- `Changes you made to given task`

### CODE REVIEW

**Comment by opeyemipeter1759** [comment]
@adaugochi kindly review my PR

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 1]
```diff
@@ -19,24 +19,23 @@ SELECT * FROM registeredPassengers;
 -- Creating the Passenger trip details table
 CREATE TABLE tripDetails (
  id INT NOT NULL AUTO_INCREMENT,
- passengerId INT NOT NULL,
- passengerClass INT NOT NULL,
- passengerTicketNo VARCHAR(20) NOT NULL,
+ passengerId INT NOT NULL Default(1),
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:22]: This shouldn't have a default value. There is no need for that

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 2]
```diff
@@ -19,24 +19,23 @@ SELECT * FROM registeredPassengers;
...
  FOREIGN KEY(passengerId) REFERENCES registeredPassengers(id)
 );
-
 SELECT * FROM tripDetails;
 
 -- Creating Accident case table
 CREATE TABLE accidentCases(
  id int not null auto_increment,
- passengerId int not null,
+ passengerId int not null default(1) ,
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:38]: same here too

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 3]
```diff
@@ -45,3 +44,29 @@ CREATE TABLE accidentCases(
...
+	select count(*) from accidentCases where accidentStatus > 0;
+	-- Answers => 342
+
+# 2. total number of unsurvived passengers
+  SELECT COUNT(*) FROM accidentCases WHERE accidentStatus = 0;
+  -- Answers => 549
+
+# 3. Name and sex of passengers under the age of 27 that embarked at Queenstown and Cherbourg
+	SELECT registeredPassengers.fullName, registeredPassengers.sex from registeredPassengers
+	inner join tripDetails on passengerId =tripDetails.passengerId where age  < '27'and embarkationPoint
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:59]: you should specify the table_name.column_name for column names that appear on the same tables you are joining

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 4]
```diff
@@ -45,3 +44,29 @@ CREATE TABLE accidentCases(
...
+	SELECT count(*) from accidentCases left join tripdetails on accidentcases.id = tripdetails.id where accidentStatus > 0 and embarkationPoint = 'S';
+    -- Answers => 218;
+
+    
+   # 5. ID, Name and Total Number of passengers who paid a fare greater than $100 and above the age of 35 had siblings or spouses on board 
+    SELECT COUNT(*) FROM registeredpassengers
+	LEFT JOIN tripdetails ON registeredpassengers.id = tripdetails.id
+	WHERE tripFare > 100 AND age > 35 AND numberOfSiblingsSpouse > 0; 
+	
+    -- Answers => 7;
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:72]: The total number of passengers is not 7

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 1 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:22]: the CSV file couldn't import correctly until I added the default value.

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 4 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:72]: @adaugochi, I have tried working on this. but I am still getting the same answer  7. 
for the record: the total number of data on all the imported tables is 891, peradventure there is an update on the CSV file i imported

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: @adaugochi kindly Review

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 1 above)
Comment (adaugochi) [reply submissions/opeyemiPeter/database/transportation-company.sql]: Hmmm, kindly specify the length of INT then. E,g INT(11)

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 4 above)
Comment (adaugochi) [reply submissions/opeyemiPeter/database/transportation-company.sql]: Ok, I will check this again

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: image
image

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 3 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: done. thanks

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 1 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: alright

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 5]
```diff
@@ -41,7 +41,43 @@ CREATE TABLE accidentCases(
...
+    ( "Q%"| "C%"));
+    
+# 4. Number of passengers who embarked at Southampton and survived
+	SELECT count(*) from accidentCases left join tripdetails on accidentcases.id = tripdetails.id where accidentStatus > 0 and embarkationPoint = 'S';
+    -- Answers => 218;
+
+# 5. 
+select registeredPassengers.id, registeredPassengers.fullName, COUNT(*) OVER () AS total_count
+from registeredPassengers 
+join tripDetails on registeredPassengers.id = tripDetails.id
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:72]: Your query is wrong here. Kindly go through the query I figure out what you are not doing correctly

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 5 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:72]: Good evening @adaugochi , i have tried every possible way to query the database to arrive at a diffferent answer but it has always given either 7(whenever i use tripsdetail.id) or 0(when i use tripsdetail.passenger.Id) as a result. i have consulted other colleagues and within and outside the codecamp about this and I tried every solution provided by them still resulting to the same result. 
if these answers(0 & 7) are not correct, then i suppose the data database will be the problem. here is the result using the different ways to querying the table.
image

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 6]
```diff
@@ -41,7 +41,43 @@ CREATE TABLE accidentCases(
...
+
+# 3. Name and sex of passengers under the age of 27 that embarked at Queenstown and Cherbourg
+	SELECT fullName, sex from registeredPassengers
+	inner join tripDetails on 
+    registeredpassengers.id = tripDetails.passengerId 
+    where (age  < '27') and (embarkationPoint = 
+    ( "Q%"| "C%"));
+    
+# 4. Number of passengers who embarked at Southampton and survived
+	SELECT count(*) from accidentCases left join tripdetails on accidentcases.id = tripdetails.id where accidentStatus > 0 and embarkationPoint = 'S';
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:66]: it supposes to be `accidentcases.passengerId = tripdetails.passengerId`. It is that `passengerId` that exist on both table that makes them have a relationship

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 5 above)
Comment (adaugochi) [reply submissions/opeyemiPeter/database/transportation-company.sql]: 0 or 7 is not the answer. Though 7 is close to it. I have checked your query and the join statement is ok. Maybe some data are missing when importing. Kindly drop the database and run all query again for the last time

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 6 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:66]: @adaugochi , i have changed it to the passengerId, thanks

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 5 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: I have imported the tables again and I still have 891 as the total number of data, but few data are missing in the cabin from trips tables and age from passengers table. i suppose that is not abnormal.
and while I use passengerId I get 0 as the answer and when I use 'id' I get  7 as the answer.

@adaugochi

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 7]
```diff
@@ -10,38 +10,82 @@ CREATE TABLE registeredPassengers (
...
+where tripFare > 100 and age > '35' and numberOfSiblingsSpouse != 0;
+
+select count(*) from registeredpassengers left join tripdetails
+on registeredpassengers.id = tripdetails.passengerId
+where tripFare > 100 and numberOfSiblingsSpouse != 0 and age > '35';
+    
+-- total number of passengers who paid a fare greater than $100 and 
+-- above the age of 35 with siblings or spouses on board
+SELECT COUNT(*) FROM registeredpassengers 
+inner join tripdetails on registeredpassengers.id = tripdetails.passengerId
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:83]: It is `tripDetails` not `tripdetails` 
image

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 7 above)
Comment (adaugochi) [reply submissions/opeyemiPeter/database/transportation-company.sql:83]: Oh..you already updated the table name

**Review by adaugochi — CHANGES_REQUESTED** [review]
Use `snake_case` for table names. Also, update your query to reflect the new table names

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 6 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: @adaugochi but do you know that this wont give 218 as its result but 220590.

image

**Comment by opeyemipeter1759** [comment]
@adaugochi kindly review the PR now

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 8]
```diff
@@ -6,42 +6,83 @@ SHOW DATABASES;
...
+    -- Answers => 218;
+
+# 5. Get the id, name and the total number of passengers 
+-- who paid a fare greater than $100 and above the age of 35 had siblings or spouses on board?
+select registered_passengers.id, registered_passengers.full_name, COUNT(*) OVER () AS total_count
+from registered_passengers 
+join trip_details on trip_Details.passenger_id = registered_passengers.id
+where trip_fare > 100 and age > '35' and number_of_siblings_spouse != 0;
+
+select count(*) from registeredpassengers left join tripdetails
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:73]: This table name `registeredpassengers` is not correct and this too `tripdetails`

**Review by adaugochi — CHANGES_REQUESTED** [review]
Kindly rerun your query to  be sure all changes reflected

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 8 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:73]: i have updated the table name and i have run the query again, i am getting the same result.

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:73]: image
@adaugochi

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:73]: @adaugochi  I tried re-importing the data again and increased the value of the name in the passengers' table then i run the data again, thus I got '9' as a result. 

thanks for your time.

image

**Comment by opeyemipeter1759** [comment]
@adaugochi kindly review my PR

**Comment by ToluEnny24** [comment]
@opeyemipeter1759 I really can imagine how exhausting the back and forth can be, well done. Just passing by

**Comment by opeyemipeter1759** [comment]
> @opeyemipeter1759 I really can imagine how exhausting the back and forth can be, well done. Just passing by

@ToluEnny24 thank you so much, it's all for the best.

**Comment by opeyemipeter1759** [comment]
> > @opeyemipeter1759 I really can imagine how exhausting the back and forth can be, well done. Just passing by
> 
> @ToluEnny24 thank you so much, it's all for the best.

thank you for checking up. i am grateful @ToluEnny24

**Comment by ToluEnny24** [comment]
Dont mind me i saw 30 comments and i was worried.. Its all for the best indeed.. We muuve

**Comment by opeyemipeter1759** [comment]
Yes ooo. It's was drilling but it's cool.
The database import was the main issue.

I have found my way around it.
I am just hoping I would be merged in the morning so I can move to the next
module.

**Comment by opeyemipeter1759** [comment]
@adaugochi good morning kindly check my PR now, i have made the correction you requested for

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 9]
```diff
@@ -6,42 +6,78 @@ SHOW DATABASES;
...
+	SELECT count(*) from accident_cases left join trip_details on 
+    accident_cases.passenger_id = trip_details.passenger_id where 
+    accident_status > 0 and embarkation_point = 'S';
+    -- Answers => 218;
+
+    
+-- total number of passengers who paid a fare greater than $100 and 
+-- above the age of 35 with siblings or spouses on board
+SELECT COUNT(*) FROM registered_passengers 
+inner join trip_details on registered_passengers.id = trip_details.id
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:75]: This should be `trip_details.passenger_id`. Though both will give you the same result but, that's not the right thing to do

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 10]
```diff
@@ -6,42 +6,78 @@ SHOW DATABASES;
...
+    
+# 4. Number of passengers who embarked at Southampton and survived
+	SELECT count(*) from accident_cases left join trip_details on 
+    accident_cases.passenger_id = trip_details.passenger_id where 
+    accident_status > 0 and embarkation_point = 'S';
+    -- Answers => 218;
+
+    
+-- total number of passengers who paid a fare greater than $100 and 
+-- above the age of 35 with siblings or spouses on board
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:73]: Also, the number 5 question has two parts, you only return the `total number of passengers`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 10 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:73]: oh thats true, i removed the second part, i will update it now

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 9 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:75]: i will change it now, thanks

**Comment by opeyemipeter1759** [comment]
@adaugochi kindly review now, i have made the changes required.
thanks

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 10 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:73]: @adaugochi , you can review now. thanks

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 11]
```diff
@@ -6,42 +6,79 @@ SHOW DATABASES;
...
+    accident_cases.passenger_id = trip_details.passenger_id where 
+    accident_status > 0 and embarkation_point = 'S';
+    -- Answers => 218;
 
+    
+-- 5 total number of passengers who paid a fare greater than $100 and 
+-- above the age of 35 with siblings or spouses on board
+# 5. SELECT COUNT(*) FROM registered_passengers 
+    inner join trip_details on registered_passengers.id = trip_details.id
+    where trip_fare > 100 and age > '35' and number_0f_siblings_spouse > 0;
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:76]: What are you doing from line 72 - 76

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 11 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:76]: I am getting the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:76]: I am getting the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:76]: I am getting the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board 
@adaugochi 

then next line was get the id, name and the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board.

Comment (adaugochi) [reply submissions/opeyemiPeter/database/transportation-company.sql:76]: Your code started from line 75. Also, you are doing the same thing from line 79. Kindly check your code very well

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 11 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:77]: @adaugochi  line 80 returns a table with id , full_name and total_count
image

while 75 returns the total number of passengers which is 9.
image

Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql:77]: @adaugochi kindly review the PR and confirm with the screenshot for the out

File: submissions/opeyemiPeter/database/transportation-company.sql [hunk 12]
```diff
@@ -6,42 +6,81 @@ SHOW DATABASES;
...
+-- 5 
+-- a . The total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board
+ 
+    SELECT COUNT(*) FROM registered_passengers 
+    inner join trip_details on registered_passengers.id = trip_details.id
+    where trip_fare > 100 and age > '35' and number_0f_siblings_spouse > 0; 
+    -- Anwser => 9
 
+-- b .  The id, name and the total number of passengers who paid a fare greater than $100 and above the age of 35 with siblings or spouses on board.
+    select registered_passengers.id, registered_passengers.full_name, COUNT(*) OVER () AS total_count
```
Comment (adaugochi) [inline submissions/opeyemiPeter/database/transportation-company.sql:81]: No need for the `COUNT` again here, you have already done it in `5a`

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/opeyemiPeter/database/transportation-company.sql (hunk 12 above)
Comment (opeyemipeter1759) [reply submissions/opeyemiPeter/database/transportation-company.sql]: alright. i have taken it away.

you can check it now.
@adaugochi

**Pull request merged**